homeassistant/components/heatmiser/climate.py

```diff
@@ -116,12 +118,17 @@ def target_temperature(self):
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
         temperature = kwargs.get(ATTR_TEMPERATURE)
-        self.heatmiser.hmSendAddress(self._id, 18, temperature, 1, self.serport)
+        self._target_temperature = int(temperature)
+        self.therm.set_target_temp(self._target_temperature)
 
     def update(self):
         """Get the latest data."""
-        self.dcb = self.heatmiser.hmReadAddress(self._id, "prt", self.serport)
-        low = self.dcb.get("floortemplow ")
-        high = self.dcb.get("floortemphigh")
-        self._current_temperature = (high * 256 + low) / 10.0
-        self._target_temperature = int(self.dcb.get("roomset"))
+        self.uh1._open()
+        self.dcb = self.therm.read_dcb()
+        self._current_temperature = int(self.therm.get_floor_temp())
+        self._target_temperature = int(self.therm.get_target_temp())
+        self._hvac_mode = (
+            HVAC_MODE_OFF
+            if (int(self.therm.get_current_state()) == 0)
+            else HVAC_MODE_HEAT
+        )
```

Comment:
The handling for communication issues during runtime is missing.

Reply:
Have copied pattern from media_player component to log error if device fails to update.  Would appreciate guidance on how better to do this within the HA framework.